source/module_base/spherical_bessel_transformer.h

```diff
@@ -98,106 +91,91 @@ class SphericalBesselTransformer
      */
-    void radrfft(const int l,
-                 const int ngrid,
-                 const double cutoff,
-                 const double* const in,
-                 double* const out,
-                 const int p = 0,
-                 const bool deriv = false
+    void radrfft(
+        int l,
+        int ngrid,
+        double cutoff,
+        const double* in,
+        double* out,
+        int p = 0
     ) const;
 
+
     /**
-     * @brief Performs an l-th order spherical Bessel transform via numerical integration with Simpson's rule.
+     * @brief Spherical Bessel transform via numerical integration with Simpson's rule.
      *
-     * This function computes the spherical Bessel transform F(x) -> G(y) (or G's derivative)
-     * with input values
+     * This function computes the spherical Bessel transform F(x) -> G(y) with input
      *
      *                   p
      *          in[i] = x [i] F(x[i])
      *
-     * where p <= 2 is an integer. On finish, out[j] = G(y[j]) or dG(y[j])/dy. 
-     * x & y are specified by grid_in & grid_out, respectively.
+     * where p <= 2 is an integer. On finish, out[j] = G(y[j]).
      *
      * @param[in]   l           order of the transform
-     * @param[in]   ngrid_in    size of the input array
+     * @param[in]   ngrid_in    number of the input grid points
      * @param[in]   grid_in     input grid
      * @param[in]   in          input values
-     * @param[in]   ngrid_out   size of the output array
+     * @param[in]   ngrid_out   number of the output grid points
      * @param[in]   grid_out    output grid
      * @param[out]  out         transformed values on the output grid
-     * @param[in]   p           exponent of the extra power term in input values, must not exceed 2
-     * @param[in]   deriv       if true, the derivative of the transform is computed
+     * @param[in]   p           exponent of the extra power term in input values
+     *                          (must not exceed 2)
      *
-     * @note    This function does not allocate memory for output; it must be pre-allocated.
      * @note    Even if the input grid forms a good sampling of F(x), results would still be
      *          inaccurate for very large y values (y*dx ~ pi) because the oscillation of
      *          j_l(y*x) in this case is poorly sampled, in which case Simpson's 1/3 rule
      *          could be a bad approximation.
      * @note    p is restricted to p <= 2 in order to avoid the situation that one has to
      *          determine x^2*F(x) at x = 0 from x[i]^p*F(x[i]).
+     *
      */
-    void direct(const int l,
-                const int ngrid_in,
-                const double* const grid_in,
-                const double* const in,
-                const int ngrid_out,
-                const double* const grid_out,
-                double* const out,
-                const int p = 0,
-                const bool deriv = false
+    void direct(
+        int l,
```

Comment:
why const has been removed?

Reply:
> why const has been removed?

I'm inclined to the opinion that it might, for the sake of brevity, be better to have less const qualifiers for arguments passed by value in an interface, because they convey no actual information to the users of this class. When a user uses this function, the ```const``` in ```const int p = 0``` above does not tell the user any information on his side because his variable passed to ```p``` will get copied and not modified anyway.

Reply:
There might be one case,  suppose developer A want to pass "int &p" into the function and modify it, however, developer A accidently wrote the code as "int &p", will it be difficult for other developer B to observe this bug without adding "const"?

Reply:
> There might be one case, suppose developer A want to pass "int &p" into the function and modify it, however, developer A accidently wrote the code as "int p", will it be difficult for other developer B to observe this bug when A didn't add "const"?

Oh yes this is a point which  I overlooked. I have those const qualifiers back in the latest commit.